networkx/generators/tests/test_nonisomorphic_trees.py

```diff
@@ -8,6 +8,32 @@
 from networkx.utils import edges_equal
 
 
+def test_nonisomorphic_tree_negative_order():
+    with pytest.raises(ValueError, match="order must be non-negative"):
+        nx.number_of_nonisomorphic_trees(-1)
+    with pytest.raises(ValueError, match="order must be non-negative"):
+        next(nx.nonisomorphic_trees(-1))
+
+
+def test_nonisomorphic_tree_order_0():
+    assert nx.number_of_nonisomorphic_trees(0) == 0
+    assert list(nx.nonisomorphic_trees(0)) == []
+
+
+def test_nonisomorphic_tree_order_1():
+    assert nx.number_of_nonisomorphic_trees(1) == 1
+    nit_list = list(nx.nonisomorphic_trees(1))
+    assert len(nit_list) == 1
+    G = nit_list[0]
+    assert nx.utils.graphs_equal(G, nx.empty_graph(0))
+
+
+@pytest.mark.parametrize("n", range(5))
+def test_nonisomorphic_tree_low_order_agreement(n):
+    """Ensure all the order<2 'special cases' are consistent."""
+    assert len(list(nx.nonisomorphic_trees(n))) == nx.number_of_nonisomorphic_trees(n)
```

Comment:
Nit, and it doesn't matter here, but since Ale just mentioned it it's fresh in my brain. We can avoid loading the whole list in memory.
```suggestion
    assert sum(1 for _ in nx.nonisomorphic_trees(n)) == nx.number_of_nonisomorphic_trees(n)
```